icu4c/source/i18n/number_formatimpl.cpp

```diff
@@ -389,8 +389,12 @@ NumberFormatterImpl::macrosToMicroGenerator(const MacroProps& macros, bool safe,
                 fMixedUnitLongNameHandler.getAlias(), status);
             chain = fMixedUnitLongNameHandler.getAlias();
         } else {
+            MeasureUnit unit = macros.unit;
+            if (uprv_strcmp(macros.perUnit.getType(), "none") != 0) {
```

Comment:
Prefer `utils::unitIsBaseUnit()` rather than a string comparison to `"none"`.  (I see you had copied this from pre-existing code)

Reply:
Done!